### ISSUE TITLE
Document the fact that access to cloud metadata is disabled by default

### ISSUE THREAD

**Issue by wookasz** [jupyterhub/zero-to-jupyterhub-k8s#698]
It's not immediately obvious that `iptables` are overridden with an init container, blocking access to EC2 metadata.

This should be documented somewhere. Perhaps in https://zero-to-jupyterhub.readthedocs.io/en/latest/amazon/step-zero-aws.html.

**Comment by cam72cam** [comment]
I'd be happy to write something up, can one of the maintainers assign the ticket to me?

**Comment by metonymic-smokey** [comment]
@cam72cam are you working on this issue?
@yuvipanda could I pick this issue, if no one else is working on it?

Thanks!

**Comment by consideRatio** [comment]
@metonymic-smokey as it has passed long time, you can certainly work on this! :heart:!

Note that I think this is a Kubernetes issue in general, at least on GKE as well. I don't understand all this so well either other than we block network access from our user pods to avoid an issue.

**Comment by metonymic-smokey** [comment]
@consideRatio [this page](https://zero-to-jupyterhub.readthedocs.io/en/v0.5-doc/security.html) in the Security docs touches upon something similar in the Cloud Metadata server security section. It has a brief on how to run `iptables` in an init container, which could potentially set specific firewall rules for the user pod.

For a start, I've added a brief note about init containers overriding `iptables` and linked to the page referred above. I need a little more direction on what more I could add.

Another issue with the page: The [page](https://zero-to-jupyterhub.readthedocs.io/en/latest/amazon/step-zero-aws.html) referred above has some commands misaligned with their code blocks. I've tried to fix that here too. 

Thanks!
